PHPC-2638 and PHPC-2630: Bump to libmongoc 2.1.2 and libmongocrypt 1.16.0

jmikola opened the pull request:
https://jira.mongodb.org/browse/PHPC-2638
https://jira.mongodb.org/browse/PHPC-2630

The libmongocrypt upgrade is necessary due to a collision between "mlib" headers in libmongoc and libmongocrypt (addressed in [MONGOCRYPT-845](https://jira.mongodb.org/browse/MONGOCRYPT-845)).

jmikola committed:
PHPC-2611: Bump libmongoc to 2.1.2

jmikola committed:
PHPC-2626: Bump libmongocrypt to 1.16.0

This pulls in changes from MONGOCRYPT-845, which are necessary to produce a static build with both libmongoc 2.1.x and libmongocrypt (due to a conflict in the "mlib" headers).

jmikola committed:
Update SBOM for libmongoc 2.1.2 and libmongocrypt 1.16.0

alcaeus commented on .evergreen/config/templates/build/build-libmongoc.yml:
```diff
@@ -10,7 +10,7 @@
       - func: "compile driver"
         vars:
           PHP_VERSION: "%phpVersion%"
-          LIBMONGOC_VERSION: "2.0.2"
```
Reminder to regenerate config as you've only updated the template

alcaeus commented on src/LIBMONGOCRYPT_VERSION_CURRENT:
```diff
@@ -1 +1 @@
-1.14.1
+1.15.2-20250912+git03cba710f7
```
Do we depend on the development version of libmongocrypt 1.15.2? Would it make sense to wait for a release?

jmikola replied:
Originally opened this before the release. It has since been bumped to the 1.15.2 tag, though.

jmikola committed:
Update error expectations for libmongoc 2.1

jmikola committed:
Bump libmongoc and libmongocrypt versions for system library testing

jmikola committed:
Update libmongocrypt mlib path for PECL packaging

GromNaN commented on tests/manager/manager-ctor-006.phpt:
```diff
@@ -12,7 +12,7 @@ echo throws(function() {
+--EXPECTF--
 OK: Got MongoDB\Driver\Exception\InvalidArgumentException
-Failed to parse MongoDB URI: 'invalid'. Invalid URI Schema, expecting 'mongodb://' or 'mongodb+srv://'.
+Failed to parse MongoDB URI: 'invalid'. Invalid URI%s
```
Noted the new exception message is
> Failed to parse MongoDB URI: 'invalid'. Invalid URI, no scheme part specified.

Why not update the expected output instead of adding this place-holder?

jmikola replied:
Good question. I must have done this during an earlier iteration of this PR when I was testing against both libmongoc and 2.0.x and 2.1.x, but there is no reason not to use the newer pattern since we require 2.1.2+.

GromNaN commented on .gitmodules:
```diff
@@ -1,7 +1,8 @@
-	url = https://github.com/mongodb/mongo-c-driver.git
+	url = https://github.com/jmikola/mongo-c-driver.git
 	ignore = untracked
+	branch = 2.1.2-handshake
```
Is the fix you added on [your branch](https://github.com/jmikola/mongo-c-driver/commits/2.1.2-handshake/) necessary for this PR?

jmikola replied:
I'm still attempting to isolate the OOM compiler error in order to provide meaningful feedback to the C team. It's likely this is the result of ludicrous macro expansion in their BSON DSL (see `mongoc-handshake.c` in https://github.com/mongodb/mongo-c-driver/commit/f142c6138321d186f19fa2735662337e7c0aafe9).

jmikola replied:
Quoting [How to: Enable a 64-Bit, x64 hosted MSVC toolset on the command line](https://learn.microsoft.com/en-us/cpp/build/how-to-enable-a-64-bit-visual-cpp-toolset-on-the-command-line?view=msvc-170):

> If you run into memory limits when you use the 32-bit tools, try the 64-bit tools.

Hat tip to @kevinAlbs.

jmikola committed:
Cross-compile 32-bit Windows builds using 64-bit toolset

This should avoid memory limits when using 32-bit tools, which is relevant for libmongoc sources using the BSON DSL API.
If you run into memory limits when you use the 32-bit tools, try the 64-bit tools.
